utils/oc.py

```diff
@@ -566,3 +577,11 @@ def cleanup(self):
         for oc in self.oc_map.values():
             if oc:
                 oc.cleanup()
+
+
+class OCFail:
```

Comment:
i'm totally fine with the simple approach. however, we will probably generalize this class fairly soon -
in my view, it's more the status of OC rather then a failure. it can also be a success or "nothing" (cluster skipped for example).
the status will indicate what is the proper logging level, return code of the integration, etc.
i'm +1 to iterate over it in future PRs, but just wanted to share my thoughts

Reply:
So remove the check the in openshift_base.py class for oc and handle this as Amadaor mentioned for `None` in the oc.py class?

Reply:
I'll let @apahim answer

Reply:
not sure I got the question...

Reply:
Comments above from your analysis of handling all error paths.

Reply:
@maorfr I took some updated suggestions from @apahim which I think round out this solution even better. Let me know your thoughts.